astropy/units/quantity_helper/function_helpers.py

```diff
@@ -344,6 +343,32 @@ def concatenate(arrays, axis=0, out=None):
     return (arrays,), kwargs, unit, out
 
 
+@dispatched_function
+def block(arrays):
+    # We need to override block since the numpy implementation can take two
+    # different paths, one for concatenation, one for creating a large empty
+    # result array in which parts are set.  Each assumes array input and
+    # cannot be used directly.  Since it would be very costly to inspect all
+    # arrays and then turn them back into a nested list, we just copy here the
+    # second implementation, np.core.shape_base._block_slicing, since it is
+    # shortest and easiest.
```

Comment:
Maybe mention the license for NUMPY is in https://github.com/astropy/astropy/blob/master/licenses/NUMPY_LICENSE.rst (since we are copying code)

Reply:
Or maybe this could go at the top of the file somewhere since this is the case for several functions here

Reply:
Good, point.  There are two locations, and I now rewrote the top to read
```
# Licensed under a 3-clause BSD style license. See LICENSE.rst except
# for parts explicitly labelled as being (largely) copies of numpy
# implementations; for those, see licenses/NUMPY_LICENSE.rst.
```